plugins/goyave/souin.go

```diff
@@ -0,0 +1,126 @@
+package goyave
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/darkweak/souin/api"
+	"github.com/darkweak/souin/cache/coalescing"
+	"github.com/darkweak/souin/configurationtypes"
+	"github.com/darkweak/souin/plugins"
+	"github.com/darkweak/souin/rfc"
+	"goyave.dev/goyave/v4"
+)
+
+const (
+	getterContextCtxKey key = "getter_context"
+)
+
+var (
+	DefaultConfiguration = Configuration{
+		DefaultCache: &configurationtypes.DefaultCache{
+			TTL: configurationtypes.Duration{
+				Duration: 10 * time.Second,
+			},
+		},
+		LogLevel: "info",
+	}
+	DevDefaultConfiguration = Configuration{
+		API: configurationtypes.API{
+			BasePath: "/souin-api",
+			Prometheus: configurationtypes.APIEndpoint{
+				Enable: true,
+			},
+			Souin: configurationtypes.APIEndpoint{
+				Enable: true,
+			},
+		},
+		DefaultCache: &configurationtypes.DefaultCache{
+			Regex: configurationtypes.Regex{
+				Exclude: "/excluded",
+			},
+			TTL: configurationtypes.Duration{
+				Duration: 5 * time.Second,
+			},
+		},
+		LogLevel: "debug",
+	}
+)
+
+// SouinGoyaveMiddleware declaration.
+type (
+	key                   string
+	SouinGoyaveMiddleware struct {
+		plugins.SouinBasePlugin
+		Configuration *Configuration
+		bufPool       *sync.Pool
+	}
+	getterContext struct {
+		next goyave.Handler
+		rw   http.ResponseWriter
+		req  *http.Request
+	}
+)
+
+func NewHTTPCache(c Configuration) *SouinGoyaveMiddleware {
+	s := SouinGoyaveMiddleware{}
+	s.Configuration = &c
+	s.bufPool = &sync.Pool{
+		New: func() interface{} {
+			return new(bytes.Buffer)
+		},
+	}
+
+	s.Retriever = plugins.DefaultSouinPluginInitializerFromConfiguration(&c)
+	s.RequestCoalescing = coalescing.Initialize()
+	s.MapHandler = api.GenerateHandlerMap(s.Configuration, s.Retriever.GetTransport())
+
+	return &s
+}
+
+func (s *SouinGoyaveMiddleware) Handle(next goyave.Handler) goyave.Handler {
```

Comment:
The middleware should check if the connection has been hijacked (`response.Hijacked()`, for websocket uses for example).